### ISSUE TITLE
"No country present for an affiliation" error when `\maketitle` appears on a `titlepage`

### ISSUE THREAD

**Issue by DominikPeters** [borisveytsman/acmart#524]
The following document produces an error "No country present for an affiliation".

```tex
\documentclass{acmart}

\title{Test}
\author{Test Author 1}
\affiliation{%
	\institution{Test Institution}
	\country{Testland}
}
\author{Test Author 2}
\affiliation{%
	\institution{Test Institution}
	\country{Testland}
}

\begin{document}
	\begin{titlepage}
		\maketitle
	\end{titlepage}
\end{document}
```

This error occurs when there are at least 2 authors and the `\maketitle` command appears inside a group (in this example, a titlepage, but wrapping it in `\bgroup` and `\egroup` results in the same error). Using titlepage is suggested in the style for the ACM EC conference.

**Comment by DominikPeters** [comment]
The problem goes away if we add a call to `\@ACM@resetaffil` at the end of `\@ACM@checkaffil` after line 7100. But I don't understand the code well enough to say why it fixes it or whether that's a sensible way of doing it.
https://github.com/borisveytsman/acmart/blob/2dce10773e9f1d4d9f1015cf854963f88d96bcbc/acmart.dtx#L7076-L7101

**Comment by borisveytsman** [comment]
`\titlepage` is not supported in acmart

**Comment by DominikPeters** [comment]
Makes sense, but the problem also occurs without `titlepage`, in particular whenever `\maketitle` appears in a group, for example the following document also gives the "No country present for an affiliation" error:
```tex
\documentclass{acmart}

\title{Test}
\author{Test Author 1}
\affiliation{%
	\institution{Test Institution}
	\country{Testland}
}
\author{Test Author 2}
\affiliation{%
	\institution{Test Institution}
	\country{Testland}
}

\begin{document}
	{
		\maketitle
	}
\end{document}
```

This makes me think it could be a bug, such as a missing `\@ACM@resetaffil` somewhere.
